public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,316 @@
+  FieldArray,
+  Icon,
+  IconButton,
+  Input,
+  InputControl,
+  Label,
+  Select,
+  Tooltip,
+  useStyles2,
+} from '@grafana/ui';
+
+type Props = { readOnly: boolean };
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  heading: css`
+    font-size: ${theme.typography.h5.fontSize};
+    font-weight: ${theme.typography.fontWeightRegular};
+  `,
+  // set fixed position from the top instead of centring as the container
+  // may get bigger when the for is invalid
+  removeButton: css`
+    margin-top: 25px;
+  `,
+});
+
+export const TransformationsEditor = (props: Props) => {
+  const { control, formState, register, setValue, watch, getValues } = useFormContext();
+  const { readOnly } = props;
+  const [keptVals, setKeptVals] = useState<Array<{ expression?: string; mapValue?: string }>>([]);
+
+  const styles = useStyles2(getStyles);
+
+  const transformOptions = getTransformOptions();
+  return (
+    <>
+      <input type="hidden" {...register('id')} />
+      <FieldArray name="config.transformations" control={control}>
+        {({ fields, append, remove }) => (
+          <>
+            <Stack direction="column" alignItems="flex-start">
+              <div className={styles.heading}>Transformations</div>
+              {fields.length === 0 && <div> No transformations defined.</div>}
+              {fields.length > 0 && (
+                <div>
+                  {fields.map((fieldVal, index) => {
+                    return (
+                      <Stack direction="row" key={fieldVal.id} alignItems="top">
+                        <Field
+                          label={
+                            <Stack gap={0.5}>
+                              <Label htmlFor={`config.transformations.${index}.type`}>Type</Label>
+                              <Tooltip
+                                content={
+                                  <div>
+                                    <p>The type of transformation that will be applied to the source data.</p>
+                                  </div>
+                                }
+                              >
+                                <Icon name="info-circle" size="sm" />
+                              </Tooltip>
+                            </Stack>
+                          }
+                          invalid={!!formState.errors?.config?.transformations?.[index]?.type}
+                          error={formState.errors?.config?.transformations?.[index]?.type?.message}
+                          validationMessageHorizontalOverflow={true}
+                        >
+                          <InputControl
+                            render={({ field: { onChange, ref, ...field } }) => {
+                              // input control field is not manipulated with remove, use value from control
+                              return (
+                                <Select
+                                  {...field}
+                                  value={fieldVal.type}
```

Comment:
@ifrost this is to fix the bug I talked about.

Replication steps:
1. Have a transformation saved already. Refresh to clean slate the state.
2. Open the correlation and go to the transformation. Delete it
3. Add a new transformation. The select will be pre-filled with the original transformation

To see what the deal is, if you add a console.log for `field` in the `InputControl` render and `fieldVal` in the `fields.map` above, you will see the InputControl field still persists this value, as it must grab it from some state that does not get affected by `remove`. If there's a more obvious solution here, please let me know :)